ui/ui-config/pages/app/new/index.tsx

```diff
@@ -20,87 +24,96 @@ class CustomErrorHandler extends EventDispatcher<{ message: string }> {
-  }: {
-    plugins: Actions.PluginsList["plugins"];
-  } = props;
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
 
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  const client = generateClient(ctx);
+  const { plugins } = await client.request<Actions.PluginsList>(
+    "get",
+    `/plugins`,
+    {
+      includeInstalled: true,
+      includeAvailable: true,
+      includeVersions: false,
+    }
+  );
+  return { props: { plugins } };
+};
+
+const Page: NextPageWithInferredProps<typeof getServerSideProps> = ({
+  plugins,
+}) => {
   const router = useRouter();
   const { client } = useApi();
-  const [plugin, setPlugin] = useState<
-    Partial<Actions.PluginsList["plugins"][number]>
-  >({ name: "" });
-  const [cards, setCards] = useState([]);
+  const [plugin, setPlugin] = useState<Partial<PluginWithVersion>>(() => ({
+    name: "",
+  }));
+  const [cards, setCards] = useState<PluginWithVersion[]>([]);
   const [loading, setLoading] = useState(false);
   const [showSpinner, setShowSpinner] = useState(true);
-  const [installingMessage, setInstallingMessage]: [
-    string | boolean,
-    Function
-  ] = useState(false);
+  const [installingMessage, setInstallingMessage] = useState<React.ReactNode>();
 
   async function resetPluginsAndApps() {
     const { plugins: _plugins }: Actions.PluginsList = await client.request(
       "get",
       `/plugins`
     );
-    prepareCards(_plugins);
+    return prepareCards(_plugins);
   }
 
   function prepareCards(_plugins: Actions.PluginsList["plugins"]) {
-    setCards(
-      _plugins
-        .filter((p) => p.source || p.destination)
-        .sort((a, b) => (a.name > b.name ? 1 : -1))
-        .sort((a, b) => {
-          if (a.installed === b.installed) return 0;
-          if (a.installed) return -1;
-          return 1;
-        })
-    );
+    const nextCards = _plugins
+      .filter((p) => p.source || p.destination)
+      .sort((a, b) => (a.name > b.name ? 1 : -1))
+      .sort((a, b) => {
+        if (a.installed === b.installed) return 0;
+        if (a.installed) return -1;
+        return 1;
+      });
+    setCards(nextCards);
+
+    return nextCards;
   }
 
   useEffect(() => {
     prepareCards(plugins);
   }, [plugins]);
 
-  async function handleClick(plugin: Actions.PluginsList["plugins"][number]) {
-    if (loading) return;
-    // @ts-ignore
-    if (!plugin.installed) return installPlugin(plugin);
-
+  const newAppFromPlugin = async (plugin: PluginWithVersion) => {
     setPlugin(plugin);
-    if (plugin.apps?.length === 1) {
-      setLoading(true);
-      const response: Actions.AppCreate = await client.request("post", `/app`, {
-        type: plugin.apps[0].name,
-      });
-      if (response?.app) {
-        return router.push("/app/[id]/edit", `/app/${response.app.id}/edit`);
-      } else {
-        setLoading(false);
-      }
-    } else {
-      router.push(`/app/new/${plugin.name}`);
```

Comment:
it looks like this page was /trying/ to do the proper logic before.  Can you call out what the problem was?

Reply:
It was only doing the proper logic when the plugin was already installed. But if it needed to install the plugin, it would not do the same thing.